pkg/reconciler/ksvc/resources/knative_route.go

```diff
@@ -31,15 +31,27 @@ func CreateKnativeRoute(kfsvc *v1alpha1.KFService) *knservingv1alpha1.Route {
 			},
 		})
 	}
+	var kfsvcAnnotations map[string]string
+	filteredAnnotations := filter(kfsvc.Annotations, isFilteredRouteAnnotation)
+	if len(filteredAnnotations) > 0 {
```

Comment:
What's wrong with applying an empty map to the resource? Does this trigger extra reconcile loops?

Reply:
empty map does not stick and when you get back from controller cache it again gives nil, so ye this triggers extra loops.